### ISSUE TITLE
[nullsafety] --macro include() can break null safety

### ISSUE THREAD

**Issue by Gama11** [HaxeFoundation/haxe#7954]
`source/foo/Main.hx`:

```haxe
package foo;

class Main {
	static var v:Int;

	static function main() {}
}
```

```hxml
-cp source
-main foo.Main
--macro nullSafety('foo')
--macro include("", true, null, ["source"])
```

This works as expected:

```
>haxe build.hxml
source/foo/Main.hx:4: characters 2-19 : Null safety: Field "v" is not nullable thus should have an initial value.
```

However, if you swap the two `--macro` calls, null safety silently breaks / doesn't complain anymore.

**Comment by ncannasse** [comment]
I think it's by-design. The include will prevent null safety metadata for being registered in the classes.

**Comment by Gama11** [comment]
That seems highly surprising / dangerous (since it makes null safety silently do nothing). I think @ibilon ran into the same issue, and he had a case where it's tricky to fix the order as well.

How is it by design? `nullSafety()` uses `addGlobalMetadata()` under the hood, not `include()`. Or are they related?

**Comment by ncannasse** [comment]
I think addGlobalMetadata needs to be called _before_ the modules are parsed, and _include_ does load all the modules. One solution would be for include to "queue" the modules from being loaded instead.

**Comment by Gama11** [comment]
Ah, I see.. it wasn't obvious to me that `include()` immediately _does_ the inclusion when you call it (rather than doing something like your proposed solution).

**Comment by kLabz** [comment]
Was fixed in #11324 (and 4.3.x)
Edit: or even earlier?

**Comment by kLabz** [comment]
Can't repro on 4.0.0 either.
Anyway, these macro API have now been cleaned up and happen in proper order.
